ark/internal/controller/query_controller.go

```diff
@@ -477,40 +477,68 @@ func (r *QueryReconciler) finalize(ctx context.Context, query *arkv1alpha1.Query
 func (r *QueryReconciler) executeTarget(ctx context.Context, query arkv1alpha1.Query, target arkv1alpha1.QueryTarget, impersonatedClient client.Client, memory genai.MemoryInterface, tokenCollector *genai.TokenUsageCollector) ([]genai.Message, error) {
 	// Create trace based on target type with input/output at trace level
 	tracer := telemetry.NewTraceContext()
+
 	ctx, span := tracer.StartSpan(ctx, fmt.Sprintf("query.%s", target.Type),
 		attribute.String("target.type", target.Type),
 		attribute.String("target.name", target.Name),
 		attribute.String("query.name", query.Name),
 		attribute.String("query.namespace", query.Namespace),
-		attribute.String("input.value", query.Spec.Input),
 	)
 	defer span.End()
 
+	// Get input messages and marshal to JSON for comprehensive telemetry
+	var inputValue string
+
+	var err error
+	metadata := map[string]string{"targetType": target.Type, "targetName": target.Name}
+
+	// Get input messages for processing and telemetry
```

Comment:
is this actually for telemetry? or is it that if it fails we record the error to the temetry provider, cause the actual messages are logged later on as they were originally?